daemon/logic/worklog.go

```diff
@@ -242,17 +270,257 @@ func (h *inviteApproveHandler) resolve(ctx context.Context, n *observer.Notifier
 	if err != nil {
 		return nil, err
 	}
+
+	return &apitypes.WorklogResult{
+		ID:      item.ID,
+		State:   apitypes.SuccessWorklogResult,
+		Message: "User invite approved and finalized.",
+	}, nil
+}
+
+type keyringMembersHandler struct {
+	engine *Engine
+}
+
+func (keyringMembersHandler) resolveErr() string {
+	return "Error adding user(s) to keyring"
+}
+
+func (h *keyringMembersHandler) list(ctx context.Context, org *envelope.Unsigned) ([]apitypes.WorklogItem, error) {
+	projects, err := h.engine.client.Projects.List(ctx, org.ID)
 	if err != nil {
-		return &apitypes.WorklogResult{
-			ID:      item.ID,
-			State:   apitypes.ErrorWorklogResult,
-			Message: "Error approving invite: " + err.Error(),
-		}, nil
+		return nil, err
+	}
+
+	// Find all of the credential graphs in this org.
+	// Search won't return all of the keyring memberships, so we have to
+	// use it just to find all of the distinct pathexps for keyrings.
+	//
+	// The paths map takes care of eliminating multiple versions of the keyring;
+	// the subsequent List call will return all versions.
+	cgs := newCredentialGraphSet()
+	orgName := org.Body.(*primitive.Org).Name
+	paths := make(map[string]*pathexp.PathExp)
+	for _, project := range projects {
+		projName := project.Body.(*primitive.Project).Name
+		graphs, err := h.engine.client.CredentialGraph.Search(ctx,
+			"/"+orgName+"/"+projName+"/*/*/*/*", h.engine.session.AuthID())
```

Comment:
That exactly! It wouldn't be a ton of work to push this behaviour into the server, and we do this in many different places (torus ls).